pxb-tag/scripts/tag.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+BRANCH=dev # default
+TAG_SUFFIX='' # default
+
+# This script assumes that the package.json (version info) and the CHANGELOG are in the same directory.
+
+# Load the package name and current version from ./package.json
+PACKAGE=`node -p "require('./package.json').name"`
+CURRENT_VERSION=`node -p "require('./package.json').version"`
+
+# Parse the command line arguments and assign to variables
+# -b: branch (default: master)
+# -s: tag suffix (default: '')
+while getopts b:s: flag
+do
+    case "${flag}" in
+        b) BRANCH=${OPTARG};;
+        s) TAG_SUFFIX=${OPTARG};;
+    esac
+done
+
+NPM_LATEST_VERSION=`npm show ${PACKAGE} version`
+
+# Check if this is an alpha, beta, or latest package and run the appropriate publishing command
+if grep -q "alpha" <<< "$CURRENT_VERSION" || grep -q "beta" <<< "$CURRENT_VERSION";
+then
+    echo "This is an alpha or beta version - skipping tag."
+    exit 0;
+else
+    # If this is not the master branch, do not do any 'latest' releases
+    if ! [ $BRANCH == "master" ];
+    then
+        echo "This is not the master branch - skipping tag."
+        exit 0;
+    fi
+
+    # If this is the master branch (or running locally without a -b flag), allow publishing a latest package
+    echo "Tagging new latest";
+
+    # Create tag-specific CHANGELOG, catch error.
+    PARSE_SCRIPT_RESPONSE=`node node_modules/@pxblue/tag/scripts/parse-changelog.js $CURRENT_VERSION`
+    echo $PARSE_SCRIPT_RESPONSE;
+    if grep -q "Error" <<< "$PARSE_SCRIPT_RESPONSE"
```

Comment:
What if the changelog contains the word Error, e.g. "Fixed login error". Would it be better to return an exit code in the parse script itself instead of doing it based on the return value here?

Reply:
> Is there logic anywhere in here to _not_ tag a master branch if the code going in is not a new latest version? Thinking for repos that have multiple packages in them, or the off chance that we push something into master that doesn't need to get published/tagged.

This script is only designed to work for the master branch right now.  Worst case scenario, if it tries to make a tag for something that has already been released, it will throw an error that says something along the lines of "current tag already released"

> What was the issue you were running into with using NPX?

I cannot call the parse-changelog node script with the `tag.sh`.  It's because of relative paths, but even whenever I started to try to use absolute paths, still no luck.  If there was only 1 script file, this would be a lot easier but I don't want to do changelog parsing logic in bash.

Reply:
So I still have the same question if the string "Error" is included somewhere in a legitimate changelog entry does that cause an issue here. Is it safer to return exit codes in the parsing script itself?

Reply:
Yes you're right, I've updated the tag.sh script to read read exits codes from the parse script.